arbnode/mel/runner/logs_fetcher.go

```diff
@@ -0,0 +1,124 @@
+package melrunner
```

Comment:
This looks great! Could we add a few simple unit tests just so it doesn't break in the future?